File: src/main/java/org/gdg/withgo/RegisterHandler.java
package org.gdg.withgo;

import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.RequestHandler;
import org.gdg.withgo.model.account.RegisterRequest;
import org.gdg.withgo.service.Postgresql;

import java.sql.*;

public class RegisterHandler implements RequestHandler<RegisterRequest, Boolean> {

    public Boolean handleRequest(RegisterRequest input, Context context){
        context.getLogger().log("Input : "+ input);

        try(Connection connection = Postgresql.create()) {
            input.assertFields();
            PreparedStatement stmt = connection.prepareStatement("select count(*) from account where phone=? OR email=?");
            stmt.setString(1, input.getPhone());
            stmt.setString(2, input.getEmail());
            ResultSet resultSet = stmt.executeQuery();
            if(resultSet.next()){
                int count = resultSet.getInt(1);
                context.getLogger().log(String.format("match count : %d ", count));
                if(count > 0){
                    return false;
                }
            }
            PreparedStatement insertStmt = connection.prepareStatement("insert into account (name, email, phone, password) values(?, ?, ?, ?)");
            insertStmt.setString(1, input.getName());
            insertStmt.setString(2, input.getEmail());
            insertStmt.setString(3, input.getPhone());
            insertStmt.setString(4, input.getPassword());
            int res = insertStmt.executeUpdate();
            if(res == 1){
                return true;
            }
        } catch (AssertionError e){
            context.getLogger().log(e.getMessage());
        } catch (SQLException e) {
            context.getLogger().log(e.getSQLState());
        }
        return false;
    }
}

File: src/main/java/org/gdg/withgo/handler/account/RegisterHandler.kt
package org.gdg.withgo.handler.account

import com.amazonaws.services.lambda.runtime.Context
import com.amazonaws.services.lambda.runtime.RequestHandler
import org.gdg.withgo.data.model.account.RegisterRequest
import org.gdg.withgo.data.repository.postgre.AuthRepository
import java.lang.AssertionError

class RegisterHandler : RequestHandler<RegisterRequest, Boolean> {

    private val authRepository: AuthRepository = AuthRepository()

    override fun handleRequest(input: RegisterRequest, context: Context): Boolean {
        context.logger.log("Register : $input")

        return try {
            input.assertFields()
            val throwable = authRepository.register(input.email, input.name, input.password, input.phone).blockingGet()
            if (throwable != null) {
                throw throwable
            }
            context.logger.log("Login success")
            true
        } catch (e: AssertionError) {
            context.logger.log("assert error ${e.message}")
            false
        } catch (e: Throwable) {
            context.logger.log("error occuard ${e.message}")
            false
        }
    }
}
